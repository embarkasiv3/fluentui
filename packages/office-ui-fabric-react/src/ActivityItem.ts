export * from './packages/react-fundamentals/ActivityItem';
