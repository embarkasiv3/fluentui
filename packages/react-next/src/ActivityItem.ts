import './version';
export * from '@fluentui/react/lib/ActivityItem';
