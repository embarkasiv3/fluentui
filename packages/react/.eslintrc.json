{
  // react-internal has some overrides we need to mirror
  "extends": ["../react-internal/.eslintrc"],
  "root": true
}
