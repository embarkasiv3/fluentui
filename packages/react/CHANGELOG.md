# Change Log - @fluentui/react

This log was last generated on Thu, 19 Mar 2020 17:52:34 GMT and should not be manually modified.
