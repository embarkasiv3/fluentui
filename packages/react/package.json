{
  "name": "@fluentui/react",
  "version": "8.0.0",
  "description": "Reusable React components for building web experiences.",
  "main": "lib-commonjs/index.js",
  "module": "lib/index.js",
  "typings": "lib/index.d.ts",
  "sideEffects": [
    "lib/version.js"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui"
  },
  "license": "MIT",
  "scripts": {
    "build": "just-scripts build",
    "bundle": "just-scripts bundle",
    "clean": "just-scripts clean",
    "code-style": "just-scripts code-style",
    "codepen": "node ../../scripts/local-codepen.js",
    "just": "just-scripts",
    "lint": "just-scripts lint",
    "start": "just-scripts dev:storybook",
    "start:legacy": "yarn workspace @fluentui/public-docsite-resources start",
    "start-test": "just-scripts jest-watch",
    "test": "just-scripts test",
    "update-snapshots": "just-scripts jest -u",
    "mf": "just-scripts mf"
  },
  "devDependencies": {
    "@fluentui/eslint-plugin": "^1.0.0",
    "@types/chalk": "^2.0.0",
    "@types/enzyme": "3.10.3",
    "@types/enzyme-adapter-react-16": "1.0.3",
    "@types/glob": "^7.1.1",
    "@types/node": "^10.3.2",
    "@types/react": "16.9.42",
    "@types/react-dom": "16.9.10",
    "@types/react-test-renderer": "^16.0.0",
    "@types/webpack-env": "1.16.0",
    "@fluentui/scripts": "^1.0.0",
    "@fluentui/example-data": "^8.0.0",
    "@fluentui/jest-serializer-merge-styles": "^8.0.0",
    "@fluentui/test-utilities": "^8.0.0",
    "@fluentui/webpack-utilities": "^8.0.0",
    "@fluentui/react-conformance": "^0.2.1",
    "chalk": "^2.1.0",
    "enzyme": "~3.10.0",
    "enzyme-adapter-react-16": "^1.15.0",
    "glob": "^7.1.2",
    "jest-snapshot": "~24.9.0",
    "office-ui-fabric-core": "^11.0.0",
    "react": "16.8.6",
    "react-app-polyfill": "~1.0.1",
    "react-dom": "16.8.6",
    "react-test-renderer": "^16.3.0"
  },
  "dependencies": {
    "@fluentui/date-time-utilities": "^8.0.0",
    "@fluentui/foundation-legacy": "^8.0.0",
    "@fluentui/font-icons-mdl2": "^8.0.0",
    "@fluentui/merge-styles": "^8.0.0",
    "@fluentui/react-focus": "^8.0.0",
    "@fluentui/react-hooks": "^8.0.0",
    "@fluentui/react-shared-contexts": "^1.0.0-beta.9",
    "@fluentui/react-window-provider": "^2.0.0",
    "@fluentui/set-version": "^8.0.0",
    "@fluentui/style-utilities": "^8.0.0",
    "@fluentui/theme": "^2.0.0",
    "@fluentui/utilities": "^8.0.0",
    "@microsoft/load-themed-styles": "^1.10.26",
    "tslib": "^1.10.0"
  },
  "peerDependencies": {
    "@types/react": ">=16.8.0 <18.0.0",
    "@types/react-dom": ">=16.8.0 <18.0.0",
    "react": ">=16.8.0 <18.0.0",
    "react-dom": ">=16.8.0 <18.0.0"
  }
}
