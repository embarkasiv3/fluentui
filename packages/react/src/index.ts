import './version';
export * from '@fluentui/react-internal';
export * from './Breadcrumb';
export * from './Checkbox';
export * from './ComboBox';
export * from './DetailsList';
export * from './DocumentCard';
export * from './Dropdown';
export * from './GroupedList';
export * from './Link';
export * from './Pivot';
export * from './ShimmeredDetailsList';
export * from './Slider';
export * from './Toggle';
