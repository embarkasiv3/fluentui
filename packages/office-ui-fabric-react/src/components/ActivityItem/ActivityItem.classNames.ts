import {
  mergeStyles,
  ITheme,
  getTheme
} from '../../Styling';
import {
  memoizeFunction
} from '../../Utilities';
import { IActivityItemStyles } from './ActivityItem.Props';
import { IPersonaProps } from '../../Persona';

export interface IActivityItemClassNames {
  root?: string;
  activityContent?: string;
  activityText?: string;
  personaContainer?: string;
  activityPersona?: string;
  activityTypeIcon?: string;
  commentText?: string;
  timeStamp?: string;
}

export const getClassNames = memoizeFunction((
  styles: IActivityItemStyles,
  className: string,
  activityPersonas: Array<IPersonaProps>,
  isCompact: boolean
): IActivityItemClassNames => {
  return {
    root: mergeStyles(
      'ms-ActivityItem',
      className,
      styles.root,
      isCompact && styles.isCompactRoot
    ) as string,

    personaContainer: mergeStyles(
      'ms-ActivityItem-personaContainer',
      styles.personaContainer,
      isCompact && styles.isCompactPersonaContainer
    ) as string,

    activityPersona: mergeStyles(
      'ms-ActivityItem-activityPersona',
      styles.activityPersona,
      isCompact && styles.isCompactPersona,
      !isCompact && activityPersonas && activityPersonas.length === 2 && styles.doublePersona
    ) as string,

    activityTypeIcon: mergeStyles(
      'ms-ActivityItem-activityTypeIcon',
      styles.activityTypeIcon,
      isCompact && styles.isCompactIcon
    ) as string,

    activityContent: mergeStyles(
      'ms-ActivityItem-activityContent',
      styles.activityContent,
      isCompact && styles.isCompactContent
    ) as string,

    activityText: mergeStyles('ms-ActivityItem-activityText', styles.activityText) as string,
    commentText: mergeStyles('ms-ActivityItem-commentText', styles.commentText) as string,
    timeStamp: mergeStyles('ms-ActivityItem-timeStamp', styles.timeStamp) as string
  };
});
