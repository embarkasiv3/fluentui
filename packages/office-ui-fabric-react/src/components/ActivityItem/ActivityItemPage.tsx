import * as React from 'react';
import { ActivityItemStatus } from './ActivityItem.checklist';
import { ActivityItemBasicExample } from './examples/ActivityItem.Basic.Example';
import { ActivityItemPersonaExample } from './examples/ActivityItem.Persona.Example';
import { ActivityItemCompactExample } from './examples/ActivityItem.Compact.Example';
import { IDemoPageProps } from '../../demo/components/DemoPage.types';
import { DemoPage } from '../../demo/components/DemoPage';

const ActivityItemBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ActivityItem/examples/ActivityItem.Basic.Example.tsx') as string;
const ActivityItemPersonaExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ActivityItem/examples/ActivityItem.Persona.Example.tsx') as string;
const ActivityItemCompactExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ActivityItem/examples/ActivityItem.Compact.Example.tsx') as string;

export const ActivityItemPageProps: IDemoPageProps = {
  title: 'ActivityItem',
  componentName: 'ActivityItem',
  componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/ActivityItem',
  componentStatus: ActivityItemStatus,
  examples: [
    {
      title: 'Activity Items with Icons',
      code: ActivityItemBasicExampleCode,
      view: <ActivityItemBasicExample />
    },
    {
      title: 'Activity Items with Personas',
      code: ActivityItemPersonaExampleCode,
      view: <ActivityItemPersonaExample />
    },
    {
      title: 'Compact Activity Items',
      code: ActivityItemCompactExampleCode,
      view: <ActivityItemCompactExample />
    }
  ],
  propertiesTablesSources: [require<string>('!raw-loader!office-ui-fabric-react/src/components/ActivityItem/ActivityItem.types.ts')],
  overview: require<string>('!raw-loader!office-ui-fabric-react/src/components/ActivityItem/docs/ActivityItemOverview.md'),
  bestPractices: '',
  dos: require<string>('!raw-loader!office-ui-fabric-react/src/components/ActivityItem/docs/ActivityItemDos.md'),
  donts: require<string>('!raw-loader!office-ui-fabric-react/src/components/ActivityItem/docs/ActivityItemDonts.md'),
  isHeaderVisible: true,
};

export const ActivityItemPage = (props: { isHeaderVisible: boolean }) => (<DemoPage { ...{ ...ActivityItemPageProps, ...props } } />);
