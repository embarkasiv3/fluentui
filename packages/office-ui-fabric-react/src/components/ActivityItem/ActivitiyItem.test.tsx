import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { ActivityItem } from './ActivityItem';

const defaultProps = {
  key: 1,
  activityDescription: <span>Activity description content</span>,
  comments: <span>Comment content</span>,
  timeStamp: 'Timestamp content'
};

describe('ActivityItem', () => {
  it('renders ActivitiyItem correctly', () => {
    const component = renderer.create(<ActivityItem {...defaultProps} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});