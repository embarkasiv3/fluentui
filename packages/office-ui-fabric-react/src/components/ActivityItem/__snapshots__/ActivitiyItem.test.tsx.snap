// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityItem renders ActivitiyItem correctly 1`] = `
<div
  className=
      ms-ActivityItem
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        align-items: flex-start;
        box-sizing: border-box;
        color: #666666;
        display: flex;
        font-family: 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 12px;
        font-weight: 400;
        justify-content: flex-start;
        line-height: 17px;
      }
  style={undefined}
>
  <div
    className=
        ms-ActivityItem-activityContent
        {
          padding-bottom: 0;
          padding-left: 8px;
          padding-right: 8px;
          padding-top: 0;
        }
  >
    <span
      className=
          ms-ActivityItem-activityText
          {
            display: inline;
          }
    >
      <span>
        Activity description content
      </span>
    </span>
    <div
      className=
          ms-ActivityItem-commentText
          {
            color: #333333;
          }
    >
      <span>
        Comment content
      </span>
    </div>
    <div
      className=
          ms-ActivityItem-timeStamp
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: #666666;
            font-family: 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 10px;
            font-weight: 400;
          }
    >
      Timestamp content
    </div>
  </div>
</div>
`;
